src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java

```diff
@@ -2,106 +2,245 @@
+            "src" + File.separator + "test" + File.separator + "resources" + File.separator + "zip" +
+                    File.separator + "dataset-valid.zip");
+
+    ResponseEntity<String> response = makeHarvestingByFile(dataset, null);
+    assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
+    assertNotNull(response.getBody());
+    assertTrue(response.getBody().contains("\"dataset-id\":\"1\""));
+  }
+
+  @Test
+  public void harvestDatasetWithFile_withXsltFile_expectStatus_accepted() {
+    FileSystemResource dataset = new FileSystemResource(
+            "src" + File.separator + "test" + File.separator + "resources" + File.separator + "zip" +
+                    File.separator + "dataset-valid-with-xslt-file.zip");
+    FileSystemResource xsltFile = new FileSystemResource(
+            "src" + File.separator + "test" + File.separator + "resources" + File.separator + "zip" +
+                    File.separator + "xslt-file.xslt");
+
+
+    ResponseEntity<String> response = makeHarvestingByFile(dataset, xsltFile);
+    assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
+    assertNotNull(response.getBody());
+    assertTrue(response.getBody().contains("\"dataset-id\":\"1\""));
   }
 
   @Test
-  public void harvestDatasetWithUrl_expectStatus_accepted() throws Exception {
+  public void harvestDatasetWithUrl_expectStatus_accepted() {
 
     Path datasetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid.zip");
     assertTrue(Files.exists(datasetPath));
 
-    mvc.perform(post("/dataset/{name}/harvestByUrl", "my-data-set")
-            .param("country", ITALY.xmlValue())
-            .param("language", IT.xmlValue())
-            .param("url", datasetPath.toUri().toString()))
-        .andExpect(status().isAccepted());
+    HttpHeaders requestHeaders = new HttpHeaders();
+    //As a request it is possible to upload a xsltFile, even if we don't want to include it,
+    //hence we set content ty as multipart_form_data
+    requestHeaders.setContentType(MediaType.MULTIPART_FORM_DATA);
+    MultiValueMap<String, String> body
+            = new LinkedMultiValueMap<>();
+    body.add("url", datasetPath.toUri().toString());
+    body.add("country", ITALY.xmlValue());
+    body.add("language", IT.xmlValue());
+
+    ResponseEntity<String> response =
+            testRestTemplate.postForEntity(getBaseUrl() + "/dataset/{name}/harvestByUrl",
+                    new HttpEntity<>(body, requestHeaders), String.class, "testDataset");
+
+    assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
+    assertNotNull(response.getBody());
+    assertTrue(response.getBody().contains("\"dataset-id\":\"1\""));
   }
 
+  @Test
+  public void harvestDatasetWithUrl_withXsltFile_expectStatus_accepted() {
+
+    Path datasetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid-with-xslt-file.zip");
+    assertTrue(Files.exists(datasetPath));
+    FileSystemResource xsltFile = new FileSystemResource(
```

Comment:
What is this xslt file, what does it do? Should it have an indicative naming?

Reply:
I have changed the name for what is it used for